src/lxc/commands.c

```diff
@@ -227,51 +227,33 @@ static int lxc_cmd_rsp_send(int fd, struct lxc_cmd_rsp *rsp)
 static int lxc_cmd_send(const char *name, struct lxc_cmd_rr *cmd,
 			const char *lxcpath, const char *hashed_sock_name)
 {
-	int client_fd;
+	int client_fd, saved_errno;
 	ssize_t ret = -1;
 
 	client_fd = lxc_cmd_connect(name, lxcpath, hashed_sock_name, "command");
-	if (client_fd < 0) {
-		if (client_fd == -ECONNREFUSED)
-			return -ECONNREFUSED;
-
+	if (client_fd < 0)
 		return -1;
-	}
 
 	ret = lxc_abstract_unix_send_credential(client_fd, &cmd->req,
 						sizeof(cmd->req));
-	if (ret < 0 ) {
-		if (errno == EPIPE) {
-			close(client_fd);
-			return -EPIPE;
-		}
-		close(client_fd);
-
-		return -1;
-	}
-
-	if ((size_t)ret != sizeof(cmd->req)) {
```

Comment:
This would be different in the new code where `errno` would be `0` AFAICT, although I'm not sure this can actually happen when sending. If the message is too long `sendmsg()` (used by `lxc_abstract_unix_send_credential()`) would return `-1` with `errno=EMSGSIZE` anyway. Can a `sendmsg()` actually succeed with less than requested the length?
Anyway, the new code would have to set `errno` explicitly for this case as well.

The `send()` case below is similar.

Reply:
Something I wouldn't rely on.

Reply:
Done!